Jellyfin.Plugin.Bookshelf/Providers/ComicBookInfo/ComicBookInfoProvider.cs

```diff
@@ -0,0 +1,236 @@
+using System;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using SharpCompress.Archives.Zip;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Providers;
+using MediaBrowser.Model.IO;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Jellyfin.Plugin.Bookshelf.Providers.ComicBookInfo
+{
+    public class ComicBookInfoProvider : IComicFileProvider
+    {
+        private readonly ILogger<ComicBookInfoProvider> _logger;
+
+        private readonly IFileSystem _fileSystem;
+
+        public ComicBookInfoProvider(IFileSystem fileSystem, ILogger<ComicBookInfoProvider> logger)
+        {
+            _fileSystem = fileSystem;
+            _logger = logger;
+        }
+
+        public async Task<MetadataResult<Book>> ReadMetadata(ItemInfo info, IDirectoryService directoryService, CancellationToken cancellationToken)
+        {
+            var path = GetComicBookFile(info.Path)?.FullName;
+
+            if (path is null)
+            {
+                _logger.LogError("Could not load Comic for {Path}", info.Path);
+                return new MetadataResult<Book> { HasMetadata = false };
+            }
+
+            try
+            {
+                using Stream stream = File.OpenRead(path);
+                // not yet async: https://github.com/adamhathcock/sharpcompress/pull/565
+                using var archive = ZipArchive.Open(stream);
+
+                if (archive.IsComplete)
+                {
+                    var volume = archive.Volumes.First();
+                    if (volume.Comment != null)
+                    {
+                        var jsonStream = new MemoryStream(Encoding.UTF8.GetBytes(volume.Comment));
+                        var comicBookMetadata = await JsonSerializer.DeserializeAsync<ComicBookInfoFormat>(jsonStream, new JsonSerializerOptions()
```

Comment:
```suggestion
                        var comicBookMetadata = await JsonSerializer.DeserializeAsync<ComicBookInfoFormat>(jsonStream, new JsonSerializerOptions()
```

Use `JsonDefaults.Options`. NOTE: this must not be modified.

Reply:
So there is nothing I should change there?

Reply:
The property you get from `JsonDefaults.Options`must not be modified, but you should use it. It already has string number parsing enabled, so no changes needed!

Reply:
Ah, ok. Done :)